Sources/TuistCache/GraphMappers/CacheMapper.swift

```diff
@@ -73,6 +91,14 @@ public class CacheMapper: GraphMapping {
     // MARK: - GraphMapping
 
     public func map(graph: Graph) throws -> (Graph, [SideEffectDescriptor]) {
+        let graphTraverser = GraphTraverser(graph: graph)
+        let availableTargets = graphTraverser.allTargets().map(\.target.name)
+        let missingTargets = sources.filter { !availableTargets.contains($0) }
```

Comment:
nitpick: maybe a bit cleaner:
```suggestion
        let availableTargets = Set(graphTraverser.allTargets().map(\.target.name))
        let missingTargets = sources.subtracting(availableTargets)
```

Reply:
I'd then have to convert `availableTargets` to `Array` when throwing the error, so maybe for this case I'd keep it as-is. Although I agree this part alone looks much better!

Reply:
actually, went with your suggestion after all 😄